packages/react-component-library/src/components/NumberInput/NumberInput.tsx

```diff
@@ -87,6 +87,12 @@ export const NumberInput: React.FC<NumberInputProps> = ({
     setNextValue,
   } = useValue(value)
 
+  // if something changes the value, such as a call to setField in Formik
+  // then update the value of the field.
+  useEffect(() => {
```

Comment:
This could work well if:

```
  useEffect(() => {
    setCommittedValue(value)
  }, [value])
```

was place inside the `useValue` hook.

Reply:
I'll try